src/robot/libraries/Collections.py

```diff
@@ -432,6 +454,13 @@ def _index_error(self, list_, index):
         raise IndexError('Given index %s is out of the range 0-%d.'
                          % (index, len(list_)-1))
 
+    def _validate_list(self, list, position=1):
+        if not is_list_like(list):
+            raise TypeError("Expected %i. argument to be a list, got '%s' instead." % (position, type_name(list)))
+
+    def _validate_lists(self, *lists):
+        for item in lists:
+            self._validate_list(item, lists.index(item)+1)
```

Comment:
This is a clever utility that, as I commented above, can be used more. The code should, however, be changed to:

```python
for index, item in enumerate(lists, start=1):
    self._validate_list(item, index)
```